src/reducers/scrollStateHelper.js

```diff
@@ -286,9 +283,15 @@ function calculateRenderedRowRange(state, scrollAnchor) {
     updateRowHeight(state, rowIdx);
   }
 
-  state.firstRowIndex = firstViewportIdx;
-  state.firstRowOffset = firstRowOffset;
+  const { availableHeight } = scrollbarsVisibleSelector(state);
+  if (lastIndex !== undefined) {
+    // Calculate offset needed to position last row at bottom of viewport
+    // This should be negative and represent how far the first row needs to be offscreen
+    firstOffset = Math.min(availableHeight - totalHeight, 0);
+  }
 
+  state.firstRowIndex = firstViewportIdx;
```

Comment:
Technically here we could end up with a firstRowIndex which points to the row before the first and a firstRowOffset which is larger than the first row...

Since we only use this to figure out the scroll y and to pass to onScrollStart & onScrollEnd, it could be fine... but I'm going to loop back and fix, I just need to make sure adding an extra call to always lookup the first row index from the scroll y value won't be bad for perf.